lib/ethpy/ethpy/test_fixtures/deploy_hyperdrive.py

```diff
@@ -0,0 +1,242 @@
+        FixedPoint(scaled_value=int(0.04665e18)) * (fp_apr * 100)
+    )
+    return (FixedPoint(scaled_value=int(1e18)) / time_stretch).scaled_value
+
+
+def initialize_deploy_account(web3: Web3) -> LocalAccount:
+    """Initializes the local anvil account to deploy everything from.
+
+    Arguments
+    --------
+    web3 : Web3
+        web3 provider object
+
+    Returns
+    -------
+    LocalAccount
+        The LocalAccount object
+    """
+    # TODO get private key of this account programatically
+    # This is the private key of account 0 of the anvil prefunded account
+    account_private_key = "0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80"
+    account: LocalAccount = Account().from_key(account_private_key)
+    # Ensure this private key is actually matched to the first address of anvil
+    assert web3.eth.accounts[0] == account.address
+    return account
+
+
+def deploy_hyperdrive_factory(rpc_url: str, deploy_account: LocalAccount) -> tuple[Contract, Contract]:
+    """Deploys the hyperdrive factory contract on the rpc_url chain
+
+    Arguments
+    --------
+    rpc_url: str
+        The RPC URL of the chain
+    deploy_account: LocalAccount
+        The account that deploys the contracts
+
+    Returns
+    -------
+    tuple[Contract, Contract]
+        The base token contract and the factory contract respectively
+    """
+    # TODO parameterize these parameters
+    # pylint: disable=too-many-locals
+    # Initial factory settings
+    initial_variable_rate = int(0.05e18)
+    curve_fee = int(0.1e18)  # 10%
+    flat_fee = int(0.0005e18)  # 0.05%
+    governance_fee = int(0.15e18)  # 0.15%
+    max_curve_fee = int(0.3e18)  # 30%
+    max_flat_fee = int(0.0015e18)  # 0.15%
+    max_governance_fee = int(0.30e18)  # 0.30%
+    # Configuration settings
+    abi_folder = "packages/hyperdrive/src/abis/"
+
+    # Load compiled objects
+    abis, bytecodes = load_all_abis(abi_folder, return_bytecode=True)
+    web3 = initialize_web3_with_http_provider(rpc_url, reset_provider=False)
+    # Convert deploy address to checksum address
+    deploy_addr = Web3.to_checksum_address(deploy_account.address)
+
+    # Deploy contracts
+    base_token_addr, base_token_contract = deploy_contract_and_return(
+        web3,
+        abi=abis["ERC20Mintable"],
+        bytecode=bytecodes["ERC20Mintable"],
+        deploy_addr=deploy_addr,
+    )
+
+    pool_addr = deploy_contract(
+        web3,
+        abi=abis["MockERC4626"],
+        bytecode=bytecodes["MockERC4626"],
+        deploy_addr=deploy_addr,
+        args=[base_token_addr, "Delvnet Yield Source", "DELV", initial_variable_rate],
+    )
+
+    forwarder_factory_addr, forwarder_factory_contract = deploy_contract_and_return(
+        web3,
+        abi=abis["ForwarderFactory"],
+        bytecode=bytecodes["ForwarderFactory"],
+        deploy_addr=deploy_addr,
+    )
+
+    deployer_addr = deploy_contract(
```

Comment:
I went on a goose chase from not noticing the difference between `deployer_addr` and `deploy_addr`. Can you use another name? I _think_ the `deploy_addr` is the address for the account that is funding the gas for the deployment? Are they also paying the initial balance? And the `deployer_addr` is the address of the deployed contract? So would `contract_addr` be more appropriate?